RecoLocalCalo/HGCalRecProducers/plugins/HGCalRecHitWorkerSimple.cc

```diff
@@ -87,6 +91,10 @@ bool HGCalRecHitWorkerSimple::run(const edm::Event & evt, const HGCUncalibratedR
         HGCRecHitCollection & result)
 {
     DetId detid = uncalibRH.id();
+// don't produce rechit if detid is a ghost one
+    if((detid < GhostDetIdPosMax_ +1  and GhostDetIdPosMin_-1 < detid ) or (detid < GhostDetIdNegMax_+1  and GhostDetIdNegMin_-1 < detid ) )
```

Comment:
as mentioned in the main PR thread, it seems simpler and perhaps more appropriate to use just two values , let's call them rangeMask and rangeMatch and use ```detid & rangeMask == rangeMatch ``` as a check.
It's pure luck that phi is  in the least significant bits to rely on simple int range matching.